### PR TITLE
[NT-2090] Refactor Replies DataSource

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Refactors the datasource for replies in an attempt to simplify the solution over our approach to root comments.

**Note:** This PR is based off #1534. If that PR merges first then this should be rebased on the main branch, although it is simpler to merge this one first and then #1534 in a cascading manner.

# 🤔 Why

When we built the first part of threaded comments (root comments), we did all of the heavy lifting for inserting, retrying, empty and errored states, etc in the view model using signals. This introduces quite a bit of complexity and for replies we're trying a different approach.

# 🛠 How

Moved most of the code to the datasource with the expectation that, instead of modifying the entire array in the view model and emitting it to be reloaded, we only emit the changes that are necessary and manipulate the data in the datasource where we can also write tests for it and where it may be easier to reason about.

# ✅ Acceptance criteria

- [ ] Team to review approach and tests.
- [x] Currently the first page of replies should still load normally.

### CODE REVIEW

File: Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift [hunk 1]
```diff
@@ -5,25 +5,63 @@ import UIKit
...
+
+    let allComments = newComments + current
+
+    allComments.forEach { comment, project in
+      self.loadValue(comment, project: project)
+    }
+  }
+
+  // TODO: Implement and write tests
+  internal func showEmptyState() {
```
Comment (singhhari) [inline Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift]: Deleting this as there is no empty state for replies. If there are no replies, the `View Replies` button is hidden.

**Review by moderateepheezy — APPROVED** [review]

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift:83:1 | warning: (indent) Indent code in accordance with the scope level. |
| Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift:121:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift [hunk 2]
```diff
@@ -36,4 +62,152 @@ internal final class CommentRepliesDataSource: ValueCellDataSource {
       assertionFailure("Unrecognized combo: \(cell), \(value).")
     }
   }
+
+  /// Determines if a `Comment` object is present in the data source at a given `IndexPath`.
```
Comment (moderateepheezy) [inline Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift:66]: Maybe should be 
`///get a comment at indexpath if present`.
Or something similar.

Comment (singhhari) [reply Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift]: Updated.

**Review by singhhari — APPROVED** [review]
Added documentation, tests and cleaned up some unused code.

**Pull request merged**